File: internal/content/testdata/bench/csharp/median-of-two-sorted-arrays.cs
public class Solution {
    public double find_median_sorted_arrays(long[] nums1, long[] nums2) {
        if (nums1.Length > nums2.Length) {
            long[] tmp = nums1; nums1 = nums2; nums2 = tmp;
        }
        int m = nums1.Length, n = nums2.Length;
        int total = m + n;
        int lo = 0, hi = m;
        while (lo <= hi) {
            int cut1 = (lo + hi) / 2;
            int cut2 = (total + 1) / 2 - cut1;
            long l1 = (cut1 > 0) ? nums1[cut1 - 1] : long.MinValue;
            long l2 = (cut2 > 0) ? nums2[cut2 - 1] : long.MinValue;
            long r1 = (cut1 < m) ? nums1[cut1] : long.MaxValue;
            long r2 = (cut2 < n) ? nums2[cut2] : long.MaxValue;
            if (l1 <= r2 && l2 <= r1) {
                if (total % 2 == 0) {
                    return (Math.Max(l1, l2) + Math.Min(r1, r2)) / 2.0;
                } else {
                    return (double)Math.Max(l1, l2);
                }
            } else if (l1 > r2) {
                hi = cut1 - 1;
            } else {
                lo = cut1 + 1;
            }
        }
        return 0.0;
    }
}

File: internal/content/testdata/bench/rust/median-of-two-sorted-arrays.rs
fn find_median_sorted_arrays(nums1: Vec<i64>, nums2: Vec<i64>) -> f64 {
    let (a, b) = if nums1.len() > nums2.len() {
        (nums2, nums1)
    } else {
        (nums1, nums2)
    };
    let m = a.len();
    let n = b.len();
    let total = m + n;
    let mut lo: i64 = 0;
    let mut hi: i64 = m as i64;
    while lo <= hi {
        let cut1 = (lo + hi) / 2;
        let cut2 = (total as i64 + 1) / 2 - cut1;
        let l1 = if cut1 > 0 { a[(cut1 - 1) as usize] } else { i64::MIN };
        let l2 = if cut2 > 0 { b[(cut2 - 1) as usize] } else { i64::MIN };
        let r1 = if (cut1 as usize) < m { a[cut1 as usize] } else { i64::MAX };
        let r2 = if (cut2 as usize) < n { b[cut2 as usize] } else { i64::MAX };
        if l1 <= r2 && l2 <= r1 {
            if total % 2 == 0 {
                let max_l = l1.max(l2);
                let min_r = r1.min(r2);
                return (max_l + min_r) as f64 / 2.0;
            } else {
                return l1.max(l2) as f64;
            }
        } else if l1 > r2 {
            hi = cut1 - 1;
        } else {
            lo = cut1 + 1;
        }
    }
    0.0
}
